projects/js-packages/components/components/scan-report/index.tsx

```diff
@@ -0,0 +1,208 @@
+		perPage: 5,
+	};
+
+	/**
+	 * DataView default layouts.
+	 *
+	 * This property provides layout information about the view types that are active. If empty, enables all layout types (see “Layout Types”) with empty layout data.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#defaultlayouts-record-string-view
+	 */
+	const defaultLayouts: SupportedLayouts = {
+		table: {
+			...baseView,
+			fields: [ FIELD_STATUS, FIELD_TYPE, FIELD_NAME, FIELD_VERSION ],
+			layout: {
+				primaryField: FIELD_STATUS,
+			},
+		},
+		list: {
+			...baseView,
+			fields: [ FIELD_STATUS, FIELD_VERSION ],
+			layout: {
+				primaryField: FIELD_NAME,
+				mediaField: FIELD_ICON,
+			},
+		},
+	};
+
+	/**
+	 * DataView view object - configures how the dataset is visible to the user.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#view-object
+	 */
+	const [ view, setView ] = useState< View >( {
+		type: 'table',
+		...defaultLayouts.table,
+	} );
+
+	/**
+	 * DataView fields - describes the visible items for each record in the dataset.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#fields-object
+	 */
+	const fields = useMemo( () => {
+		const iconHeight = 20;
+
+		const result = [
+			{
+				id: FIELD_STATUS,
+				label: __( 'Status', 'jetpack-components' ),
+				render( { item } ) {
+					if ( item.checked ) {
+						if ( item.threats.length > 0 ) {
+							return (
+								<Tooltip
+									className={ styles.tooltip }
+									text={ __( 'Threat detected.', 'jetpack-components' ) }
+								>
+									<div className={ styles.icon }>
+										<ShieldIcon variant="warning" height={ iconHeight } />
+									</div>
+								</Tooltip>
+							);
+						}
+						return (
+							<Tooltip
+								className={ styles.tooltip }
+								text={ __(
+									'No known threats found that affect this version.',
+									'jetpack-components'
+								) }
+							>
+								<div className={ styles.icon }>
+									<ShieldIcon variant="success" height={ iconHeight } />
+								</div>
+							</Tooltip>
+						);
+					}
+					return (
+						<Tooltip
+							className={ styles.tooltip }
+							text={ __(
+								'This item was added to your site after the most recent scan. We will check for threats during the next scheduled one.',
+								'jetpack-components'
+							) }
+						>
+							<div className={ styles.icon }>
+								<ShieldIcon variant="info" height={ iconHeight } />
+							</div>
+						</Tooltip>
+					);
+				},
+			},
+			{
+				id: FIELD_TYPE,
+				label: __( 'Type', 'jetpack-components' ),
+				elements: TYPES,
+				render( { item } ) {
+					return item.type ? item.type.charAt( 0 ).toUpperCase() + item.type.slice( 1 ) : '';
+				},
+			},
+			{
+				id: FIELD_NAME,
+				label: __( 'Name', 'jetpack-components' ),
+				enableGlobalSearch: true,
+				render( { item } ) {
```

Comment:
```suggestion
				getValue( { item } ) {
```
Use `getValue` to format both the actual value and the rendered column, to ensure a missing name will be set as an empty string when sorting.